File: src/main/java/com/dersaun/apigestaocontas/domain/dtos/DivisaoDTO.java
package com.dersaun.apigestaocontas.domain.dtos;

import com.dersaun.apigestaocontas.domain.Lancamento;
import com.dersaun.apigestaocontas.domain.enums.EDivisaoTipo;
import lombok.AllArgsConstructor;
import lombok.Data;
import org.eclipse.collections.api.list.MutableList;
import org.eclipse.collections.impl.list.mutable.FastList;

import java.io.Serializable;
import java.math.BigDecimal;

@Data
@AllArgsConstructor
public class DivisaoDTO implements Serializable {

    private EDivisaoTipo tipoDivisao;

    private BigDecimal valorLancamentoOriginal;

    private MutableList<Lancamento> lancamentos;

    public DivisaoDTO() {
        this.lancamentos = new FastList<>();
    }
}

File: src/app/conta/dtos/divisao.dto.ts
import { Lancamento } from '@app/conta/models/lancamento.entity';

export enum EDivisaoTipo {
  IGUALMENTE = 1,
  DIFERENTE,
}
export class DivisaoDTO {
  tipoDivisao: EDivisaoTipo;
  valorLancamentoOriginal: number;
  lancamentos: Lancamento[] = [];
}
